File: messenger-next2/src/types/message.ts
export enum MessageType {
  Text = 'Text',
  Image = 'Image',
  File = 'File',
  Voice = 'Voice',
  Video = 'Video',
}

export interface MessageDto {
  id: string;
  senderId: string;
  receiverId: string;
  content: string;
  isRead: boolean;
  readAt?: Date;
  type: MessageType;
  attachmentUrl?: string;
  createdAt: Date;
}

export interface CreateMessageDto {
  receiverId: string;
  content: string;
  type: MessageType;
  attachmentUrl?: string;
}

export interface ConversationDto {
  userId: string;
  username: string;
  firstName: string;
  lastName: string;
  profilePicture?: string;
  lastMessage: string;
  lastMessageTime: Date;
  unreadCount: number;
  isOnline: boolean;
  lastSeen?: Date;
} 

File: MessengerApp.Core/Entities/Message.cs
using MessengerApp.Core.Entities.Base;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace MessengerApp.Core.Entities;

public class Message : BaseEntity
{
    [BsonElement("senderId")]
    public required string SenderId { get; set; }

    [BsonElement("receiverId")]
    public required string ReceiverId { get; set; }

    [BsonElement("content")]
    public required string Content { get; set; }

    [BsonElement("type")]
    public required string Type { get; set; }

    [BsonElement("attachmentUrl")]
    public string? AttachmentUrl { get; set; }

    [BsonElement("isRead")]
    public bool IsRead { get; set; }

    [BsonElement("readAt")]
    public DateTime? ReadAt { get; set; }
}

public enum MessageType
{
    Text,
    Image,
    File,
    Voice,
    Video
} 